aws/lambdas/hhvm-get-build-status/index.js

```diff
@@ -0,0 +1,91 @@
+'use strict';
+const AWS = require('aws-sdk');
+const promise = require('promise');
+const rp = require('request-promise');
+
+
+function get_version_info(event) {
+  let version = event.version;
+  if (!version) {
+    const s = (new Date()).toISOString()
+    version = s.slice(0,4)+'.'+s.slice(5, 7)+'.'+s.slice(8, 10);
+  }
+  const nightly = !! /^\d{4}(\.\d{2}){2}$/.exec(version);
+  return {
+    nightly,
+    branch: nightly ? 'master' : ('HHVM-'+version.match(/^\d+\.\d+/)[0]),
+    version: version
+  };
+}
+
+function get_distros_uri(branch) {
+  return 'https://raw.githubusercontent.com/hhvm/packaging/'
+    + branch + '/CURRENT_TARGETS';
+}
+
+async function get_distros(branch) {
+  const response = await rp(get_distros_uri(branch));
+  return response.trim().split("\n");
+}
+
+exports.handler = async (event) => {
+  const {nightly, version, branch} = get_version_info(event);
+
+  const prefix = nightly ? 'hhvm-nightly-'+version : 'hhvm-'+version;
+  let paths = {
+    'macos-sierra':
+      'homebrew-bottles/'+prefix+'.sierra.bottle.tar.gz',
+    'macos-high_sierra':
+      'homebrew-bottles/'+prefix+'.high_sierra.bottle.tar.gz',
+  };
+  if (nightly) {
+    paths.source = 'source/nightlies/'+prefix+'.tar.gz';
+    paths.source_gpg = 'source/nightlies/'+prefix+'.tar.gz.sig';
+  } else {
+    paths.source = 'source/'+prefix+'.tar.gz';
+    paths.source_gpg = 'source/'+prefix+'.tar.gz.sig';
+  }
+
+  const distros = await get_distros(branch);
+  distros.forEach(distro => {
+    const debianish = distro.match(/^(ubuntu|debian)/);
+    if (debianish !== null) {
+      paths[distro] = debianish[0] + '/pool/main/h/' +
+        (nightly ? 'hhvm-nightly/hhvm-nightly_' : 'hhvm/hhvm_') +
+        version + '-1~' + (distro.match(/[a-z]+$/)) + '_amd64.deb';
+      return;
+    }
+    // If we add a new distro kind without updating monitoring, make
+    // the monitoring fail.
+    paths[distro] = distro;
+  });
+
+  let success = true;
+  let results = { };
+  const s3 = new AWS.S3();
+  await promise.all(
+    Object.values(paths).map(path => (async function() {
```

Comment:
Do we have to do it this way? vs. something like
```
Object.values(paths).map(async (path) => {
  //...
});
```

Reply:
there's `()` at the end - I'm returning an Awaitable, not a (function():Awaitable)